ui/raidboss/data/06-ew/raid/p4s.ts

```diff
@@ -0,0 +1,704 @@
+// TODO: Safe spot callouts for Wreath of Thorns 2?
+// TODO: Better Dark Design/tether break callouts
+// TODO: Wreath of Thorns 3 strategy (1 = melee, 2 = ranged) or
+//       something more intelligent such as tracking the vulnerabilities?
+// TODO: Call out thorns soaks for Wreath of Thorns 3 (E/W?)
+// TODO: Heart Stake is tankbuster with DoT, does it need to be output differrently?
+// TODO: Wreath of Thorns 4 headmarkers and tethers
+
+export interface Data extends RaidbossData {
+  actingRole?: string;
+  decOffset?: number;
+  fleetingImpulseCounter?: number;
+  meFleetingImpulse?: number;
+}
+
+// Due to changes introduced in patch 5.2, overhead markers now have a random offset
+// added to their ID. This offset currently appears to be set per instance, so
+// we can determine what it is from the first overhead marker we see.
+// The first 1B marker in the encounter is an Elegant Evisceration? (00DA?).
+const firstHeadmarker = parseInt('00DA', 16);
+const getHeadmarkerId = (data: Data, matches: NetMatches['HeadMarker']) => {
+  // If we naively just check !data.decOffset and leave it, it breaks if the first marker is 00DA.
+  // (This makes the offset 0, and !0 is true.)
+  if (typeof data.decOffset === 'undefined')
+    data.decOffset = parseInt(matches.id, 16) - firstHeadmarker;
+  // The leading zeroes are stripped when converting back to string, so we re-add them here.
+  // Fortunately, we don't have to worry about whether or not this is robust,
+  // since we know all the IDs that will be present in the encounter.
+  return (parseInt(matches.id, 16) - data.decOffset).toString(16).toUpperCase().padStart(4, '0');
+};
+
+const triggerSet: TriggerSet<Data> = {
+  zoneId: ZoneId.AsphodelosTheFourthCircleSavage,
+  timelineFile: 'p4s.txt',
+  timelineTriggers: [
+    {
+      id: 'P4S Dark Design',
+      regex: /Dark Design/,
+      beforeSeconds: 5,
+      infoText: (_data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Stack for Puddle AOEs',
+          de: 'Stacken (Pfützen)',
+          fr: 'Packez-vous pour l\'AoE',
+          ja: 'スタック',
+          cn: '集合放置AOE',
+          ko: '중앙에 모이기',
+        },
+      },
+    },
+    {
+      id: 'P4S Kothornos Kick',
+      regex: /Kothornos Kick/,
+      beforeSeconds: 5,
+      infoText: (data, _matches, output) => {
+        if (data.role === 'tank' || data.role === 'healer')
+          return;
+        return output.text!();
+      },
+      outputStrings: {
+        text: {
+          en: 'Bait Kothornos Kick?',
+        },
+      },
+    },
+    {
+      id: 'P4S Kothornos Quake',
+      regex: /Kothornos Quake/,
+      beforeSeconds: 5,
+      infoText: (data, _matches, output) => output.text!(),
+      outputStrings: {
+        text: {
+          en: 'Bait Kothornos Quake?',
+        },
+      },
+    },
+    {
+      id: 'P4S Hemitheos\'s Water IV',
+      regex: /Hemitheos's Water IV/,
+      beforeSeconds: 5,
```

Comment:
Is there no cast for this?

Reply:
The cast is 0.7 seconds.

00:02:49.070 | Hesperos 6 begins casting  Hemitheos's Water IV on Hesperos 6 (0.70 sec) |  
00:02:50.049 | Hesperos 6 casts  Hemitheos's Water IV

Reply:
Ok, sounds good! Just asking.  :)